src/core/annotation.js

```diff
@@ -74,40 +74,53 @@ import { OperatorList } from "./operator_list.js";
 import { XFAFactory } from "./xfa/factory.js";
 
 class AnnotationFactory {
-  static createGlobals(pdfManager) {
-    return Promise.all([
-      pdfManager.ensureCatalog("acroForm"),
-      pdfManager.ensureDoc("xfaDatasets"),
-      pdfManager.ensureCatalog("structTreeRoot"),
-      // Only necessary to prevent the `Catalog.baseUrl`-getter, used
-      // with some Annotations, from throwing and thus breaking parsing:
-      pdfManager.ensureCatalog("baseUrl"),
-      // Only necessary to prevent the `Catalog.attachments`-getter, used
-      // with "GoToE" actions, from throwing and thus breaking parsing:
-      pdfManager.ensureCatalog("attachments"),
-      pdfManager.ensureCatalog("globalColorSpaceCache"),
-    ]).then(
-      ([
+  static async createGlobals(pdfManager, xref) {
+    try {
+      const [
         acroForm,
         xfaDatasets,
         structTreeRoot,
         baseUrl,
         attachments,
         globalColorSpaceCache,
-      ]) => ({
+      ] = await Promise.all([
+        pdfManager.ensureCatalog("acroForm"),
```

Comment:
nit: Maybe we could just create an array of promises without acroform, then await on the acroform one, add the objectLoader.load the array and then await on all promises.

Reply:
Good idea.
I've attempted a slightly different approach, to keep everything in one place; please take a look.